### PR TITLE
Fix case CFG no-match edge for catch-all arms

### PR DESCRIPTION
**Pull request by ewhauser**
## Summary
- record when a case arm can match any subject
- only add the synthetic no-match CFG edge when no catch-all arm exists
- add regression coverage for catch-all and no-match case liveness

## Why
The CFG always added a head-to-exit edge for case commands, even when a catch-all arm made that path impossible.

## Impact
Dataflow no longer keeps pre-case assignments artificially live or treats post-case code as reachable when every real execution must enter an arm.

## Validation
- cargo test -p shuck-semantic

### CODE REVIEW

File: crates/shuck-semantic/src/cfg.rs [hunk 1]
```diff
@@ -644,7 +646,9 @@ impl<'a> GraphBuilder<'a> {
             }
         }
 
-        self.add_edge(head, exit_block, EdgeKind::Sequential);
+        if no_match_possible {
+            self.add_edge(head, exit_block, EdgeKind::Sequential);
```
[the remark this answers is not in the document]
Comment (ewhauser) [reply crates/shuck-semantic/src/cfg.rs]: Handled in 90c054b by giving empty case arms a synthetic CFG block and adding a regression test for an empty catch-all arm followed by reachable code.
